src/pseudopeople/noise_functions.py

```diff
@@ -610,14 +610,9 @@ def ocr_corrupt(truth, corrupted_pr, rng):
     # Apply keyboard corrupt for OCR to column
     token_noise_level = configuration[Keys.TOKEN_PROBABILITY]
     rng = np.random.default_rng(seed=get_hash(f"{randomness_stream.seed}_make_ocr_errors"))
-    column = data[column_name]
-    column = column.astype(str)
-    for idx in column.index:
-        noised_value = ocr_corrupt(
-            column.loc[idx],
-            token_noise_level,
-            rng,
-        )
-        column[idx] = noised_value
 
-    return column
+    return (
+        data[column_name]
+        .astype(str)
+        .apply(ocr_corrupt, corrupted_pr=token_noise_level, rng=rng)
```

Comment:
Oof, nice catches - I'm slightly embarrassed these loops snuck through!

Reply:
Loops aren't inherently bad! In fact there is probably a loop that is slightly faster than this `apply`. The previous loop was a bottleneck because it _used Pandas indexing_ within each iteration. Eliminating that makes this fast enough that I didn't see a need to optimize further.